src/algebra/scalar.rs

```diff
@@ -415,9 +412,45 @@ impl<C: CurveGroup> MulAssign for Scalar<C> {
 // | Conversions |
 // ---------------
 
-impl<C: CurveGroup, T: Into<C::ScalarField>> From<T> for Scalar<C> {
```

Comment:
out of curiosity, why are we not still using this generic conversion implementation? it's pretty clean and seems to still capture all of the new conversion logic

Reply:
I have no idea why -- perhaps a compiler bug -- but the compiler was unable to prove to itself after the refactor that `Scalar` did not implement `Into<C::ScalarField>`; so this line was giving "duplicate implementations" errors. Not worth trying to resolve